### PR TITLE
gitserver: add Client.P4Exec method

### PR DESCRIPTION
**Pull request by unknwon**
Adds `Client.P4Exec` method for gitserver's RPC `/p4-exec`.

Subset of #17881 and extracted here for easier review, and part of #16705.

### CODE REVIEW

**Comment by ryanslade** [comment]
@unknwon Big 👍 for breaking everything out into small easily reviewable PRs

File: internal/gitserver/client.go [hunk 1]
```diff
@@ -285,6 +285,49 @@ func (c *Cmd) sendExec(ctx context.Context) (_ io.ReadCloser, _ http.Header, err
...
+		return nil, nil, err
+	}
+
+	switch resp.StatusCode {
+	case http.StatusOK:
+		return resp.Body, resp.Trailer, nil
+
+	default:
+		// Read response body at best effort
+		body, _ := ioutil.ReadAll(resp.Body)
```
Comment (ryanslade) [inline internal/gitserver/client.go:325]: Now that we're on 1.16 maybe use io.ReadAll?

File: internal/gitserver/client_test.go [hunk 2]
```diff
@@ -286,3 +287,87 @@ func TestAddrForRepo(t *testing.T) {
 		})
 	}
 }
+
+func TestClient_P4Exec(t *testing.T) {
+	root, err := ioutil.TempDir("", t.Name())
```
Comment (ryanslade) [inline internal/gitserver/client_test.go:292]: Another use of `ioutil` :)

**Review by ryanslade — APPROVED** [review]

File: internal/gitserver/client_test.go (hunk 2 above)
Comment (unknwon) [reply internal/gitserver/client_test.go:292]: Hmm, `os.TempDir` is not a drop-in replacement for `ioutil.TempDir`, and actually have different signatures, I'll leave it for later.

Comment (keegancsmith) [reply internal/gitserver/client_test.go:292]: You want `os.MkdirTemp` not `os.TempDir` :)

**Pull request merged**